iml-services/iml-device/src/virtual_device/mod.rs

```diff
@@ -290,7 +283,7 @@ fn process_actions(mut d: Device, actions: &mut collections::HashSet<Action>) ->
             actions_to_remove.insert(a.clone());
         }
     }
-    tracing::info!("Took {} actions", actions_to_remove.len());
+    tracing::trace!("Took {} actions", actions_to_remove.len());
```

Comment:
I think debug would be better. I use this rule of thumb: only static strings _may_ be good for tracing.